### DISCUSSION TITLE
Default Container Tabs per Workspace

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#887] [Ideas]
### What feature would you like to see?

Set a default container tab on each workspace. For example, opening a new tab on the "Personal" workspace will have the "Personal" container tab by default.

### What platform are you seeing the problem on?

Linux

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/163_

**Comment by Eragonfr** [comment]
This should be configurable from the three dots and on workspace creation. But it should remain optional to set a container for a workspace

**Reply by weslley75** [reply]
Like that:
![image](https://github.com/user-attachments/assets/e4a4fc74-1e7b-4dfc-a33e-c02b7a3a0f63)

**Reply by xavierchanth** [reply]
Agree, this would be nice & should remain optional. How I would use this:

- Shared history, bookmarks, etc. between the workspaces for personal usage (as it is now - with no container)
- Completely isolating my work stuff from personal stuff by automatically using a container for my "work" workspace(s)

**Reply by Eragonfr** [reply]
I wasn't thinking of isolating history, bookmarks, etc.  
Only isolating the cookies, localstorage and cache, like how firefox does it currently.  

Separating history and others too is nice, but can't that be done by switching your browser profile ?

**Reply by ju-li** [reply]
The benefit of containers vs profiles is:

1. Each profile is its own separate window
2. With containers, you can have multiple different containers in the same window. If we tie workspaces with containers, you only need to switch workspaces to go from personal to work, and back. 

This is a much smoother experience compared to changing windows

**Reply by Eragonfr** [reply]
I understand but that would also mean that you need to expand the scope of the containers work currently.
The history, settings, extensions, etc are shared between containers.
A container is only a separation of the storage available for the webpage.
I personally don't want to have a container that changes my history.
I think that this should be requested as an other feature, something like "In-place profile switch" to change your browser profile without having to change window. (It also seems way harder to implements than simply setting a default container for each workspace)
Maybe open a new feature request ? idk.

**Reply by ju-li** [reply]
> I understand but that would also mean that you need to expand the scope of the containers work currently. The history, settings, extensions, etc are shared between containers. A container is only a separation of the storage available for the webpage. I personally don't want to have a container that changes my history. I think that this should be requested as an other feature, something like "In-place profile switch" to change your browser profile without having to change window. (It also seems way harder to implements than simply setting a default container for each workspace) Maybe open a new feature request ? idk.

Just want to clarify that I'm suggesting something very simple: The option for the user to set a default container for a workspace.

There's already a container feature

There's already a workspace feature

Just want to be able to associate them with each other

This is how it's handled in Floorp, which I like. I'm hoping to migrate from Floorp because Floorp's UI is quite buggy and crashes a lot

**Reply by Eragonfr** [reply]
I believe that we are on the same page @ju-li 
But that what @XavierChanth wants is more like profile switching.

**Reply by judehunter** [reply]
for reference, there's a discussion tracking the profile switching idea: https://github.com/zen-browser/desktop/discussions/818#discussioncomment-10537541 @Eragonfr @XavierChanth

**Comment by salastro** [comment]
Any progress on this? I am currently using simple tab groups which is fine, but a bit painful.

**Reply by salastro** [reply]
How would one go on about implementing such feature? I know a moderate amount of programming, but the codebase is so huge that idk from where to begin approaching this problem.

**Comment by LarveyOfficial** [comment]
This feature would be great, this and tree-style tabs are the only thing holding me back from switching from floorp.

**Comment by Jaakkonen** [comment]
Investigated how this could be implemented and figured out that the `gBrowser.addTab` global could be wrapped to set `userContextId` to make the tab a container tab. The active workspace can be got via `ZenWorkspaces` global. Syncing the container setup and workspace setup is gonna be the more tedious part in this. But anyway seems to be really doable feature! :) If PRs are welcome I'd be happy to submit this

**Reply by salastro** [reply]
Could you direct us towards your implementation?

**Comment by mr-cheffy** [comment]
Done!

**Reply by ju-li** [reply]
Thank you! I've moved to Zen Browser and really loving it so far!

**Reply by eldelacajita** [reply]
I was so happy to find out this had been implemented that just came here to comment 🥰
